tests/sanitize/tsan.suppression

```diff
@@ -1 +1,2 @@
 race:dbms/src/Common/TiFlashMetrics.h
+race:DB::Context::setCancelTest
```

Comment:
Why is it suppressed?

Reply:
> Why is it suppressed?

Since the test_mode only make sense in UT, and we don't want to add more locks in Interpreter critical path.